### DISCUSSION TITLE
Mikro Orm config file ignored on Nestjs build using Docker

### DISCUSSION THREAD

**Discussion by alexrequelme** [mikro-orm/mikro-orm#3537] [Q&A] [answered]
**Describe the bug**
When I build the application with docker to deploy to production and run node dist/main.js I get an error that it cannot find the configuration file, but if I have defined the path in the package.json. 

**Stack trace**
```
ERROR [ExceptionHandler] MikroORM config file not found in ['./mikro-orm.config.js'] 
```

**To Reproduce**
```
"mikro-orm": {
    "useTsNode": true,
    "configPaths": [
      "./src/database/mikro-orm.config.ts",
      "./dist/database/mikro-orm.config.js"
    ]
}
```
**Expected behavior**
The configuration should load normally since the path where it is located is defined

**Additional context**

**Versions**

| Dependency | Version |
| - | - |
| node | v16.17.0 |
| typescript | 4.7.4 |
| mikro-orm | 5.4.2 |

**Comment by ttay24** [comment]
Are you using the mikro orm nest module? It takes in the config (as an actual JS object). I would assume it’s not getting bundled into your build. 

I setup a DB module in my project that imports the mikro ORM module and sets it up, and then exports it. I then import that in my various service modules 

Example:
```
import { Module } from '@nestjs/common';
import { MikroOrmModule } from '@mikro-orm/nestjs';
import { entities } from '@crew-ff-draft/entity';
import config from '../../../mikro-orm.config';

@Module({
  imports: [
    MikroOrmModule.forRoot(config),
    MikroOrmModule.forFeature({
      entities: entities,
    }),
  ],
  exports: [MikroOrmModule],
})
export class DBModule {}
```

With how you currently set it up, if you just build the nest app like normal, do you get the js config as a separate file? Or do you only have main.js? The error leads me to believe that the file doesn’t exist

**Comment by B4nan** [comment, marked as answer]
> ERROR [ExceptionHandler] MikroORM config file not found in ['./mikro-orm.config.js'] 

This kinda means you don't have package.json inside your docker image, or that it does not contain the settings as you say so -- otherwise you would see the paths you provide in that error message.

If you want help, you need to provide reproductions, this is not a reproduction. There wont be any "docker support", docker is not special, its just an environment. If things dont work, its you not configuring them properly, you should bash inside the image and verify what is actually there. This is not the right place to talk about debuging docker issues.

**Reply by alexrequelme** [reply]
That's correct, I was doing a multi-stage build and it seems that the package.json was not in the final build, thanks for the answer.
